fixtures/tests/_fixtures/test_monkeypatch.py

```diff
@@ -72,14 +83,112 @@ def test_delete_missing_attribute(self):
             fixture.cleanUp()
             self.assertFalse('new_attr' in globals())
 
+    def _check_static_or_class_method(self, name, new_method):
+        oldmethod = getattr(C, name)
+        oldmethod_inst = getattr(C(), name)
+        fixture = MonkeyPatch(
+            'fixtures.tests._fixtures.test_monkeypatch.C.%s' % name,
+            new_method)
+        with fixture:
+            getattr(C, name)()
+            getattr(C(), name)()
+
+        restored_method = getattr(C, name)
+        restored_method_inst = getattr(C(), name)
+        self.assertEqual(oldmethod, restored_method)
+        self.assertEqual(oldmethod, restored_method_inst)
+        self.assertEqual(oldmethod_inst, restored_method)
+        self.assertEqual(oldmethod_inst, restored_method_inst)
+        restored_method()
+        restored_method_inst()
+
     def test_patch_staticmethod(self):
-        oldfoo = C.foo
+        self._check_static_or_class_method('foo_staticmethod', fake_no_args)
+
+    def test_patch_staticmethod_with_staticmethod(self):
+        self._check_static_or_class_method('foo_staticmethod',
+                fake_staticmethod)
```

Comment:
This is one of the cases I worry about: we're actually patching in a descriptor as-is. Its great you've made it work, but its at least a little crazy :)